src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -88,8 +89,12 @@ void PollingResolver::RequestReresolutionLocked() {
 }
 
 void PollingResolver::ResetBackoffLocked() {
-  if (have_next_resolution_timer_) {
-    grpc_timer_cancel(&next_resolution_timer_);
+  if (next_resolution_timer_handle_.has_value()) {
```

Comment:
Suggest refactoring this block into a helper method called `MaybeCancelNextResolutionTimer()`, so that we don't need to repeat the code both here and on line 107.